### ISSUE TITLE
`py` not installed from `environment.yml`

### ISSUE THREAD

**Issue by birdsarah** [conda/conda#5193]
Steps to reproduce:

1) Create `environment.yml` like this

```yaml
name: pyenv
dependencies:
  - python ==3.6
  - py ==1.4.33
```

2) `conda env create`

3) `source activate pyenv`

4) `conda list`

RESULT:
```sh
# packages in environment at /Users/sbird/miniconda3/envs/pyenv:
#
openssl                   1.0.2k                        1  
pip                       9.0.1                    py36_1  
python                    3.6.0                         0  
readline                  6.2                           2  
setuptools                27.2.0                   py36_0  
sqlite                    3.13.0                        0  
tk                        8.5.18                        0  
wheel                     0.29.0                   py36_0  
xz                        5.2.2                         1  
zlib                      1.2.8                         3  
```

EXPECTED: I expect py to be in this list.

5) `conda install py`

```sh
Fetching package metadata ...........
Solving package specifications: .

Package plan for installation in environment /Users/sbird/miniconda3/envs/pyenv:

The following NEW packages will be INSTALLED:

    py: 1.4.33-py36_0

Proceed ([y]/n)? y
```

6) `conda list`

```sh
# packages in environment at /Users/sbird/miniconda3/envs/pyenv:
#
openssl                   1.0.2k                        1  
pip                       9.0.1                    py36_1  
py                        1.4.33                   py36_0  
python                    3.6.0                         0  
readline                  6.2                           2  
setuptools                27.2.0                   py36_0  
sqlite                    3.13.0                        0  
tk                        8.5.18                        0  
wheel                     0.29.0                   py36_0  
xz                        5.2.2                         1  
zlib                      1.2.8                         3  
```

Given steps 5 & 6 it seems like there's something weird going on with parsing my `environment.yml`.

```
$ conda --version
conda 4.3.16
```

**Comment by nehaljwani** [comment]
Interestingly, if the `environment.yaml` is changed to...

```
name: pyenv
dependencies:
  - py ==1.4.33
  - python ==3.6
```

...then it works as expected!

**Comment by nehaljwani** [comment]
This has already been fixed in https://github.com/conda/conda/pull/5103 . Will probably be available in `v4.3.17`

**Comment by kalefranz** [comment]
Try canary Sarah!

    conda config --add channels conda-canary
    conda update conda
